src/components/MainPage/MainPage.tsx

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useState, useCallback } from "react";
+import { useNavigate, Link } from "react-router-dom";
+
+import { getCats, ICat } from "../../api";
+import { Card, Spinner, CatDetailsModal } from "../shared";
+
+import Styled from "./MainPage.styled";
+
+export const MainPage = () => {
+  const [cats, setCats] = useState<ICat[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const navigate = useNavigate();
+
+  const fetchCats = useCallback(async () => {
+    const catsDisplayed = cats.length;
+    const page = catsDisplayed > 0 ? catsDisplayed / 10 : 0;
+    setIsLoading(true);
+    try {
+      const response = await getCats(page);
+      setCats((cats) => [...cats, ...response.data]);
+    } catch (error) {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  }, [cats]);
+
+  useEffect(() => {
+    fetchCats();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why did you need to add this line here? Is this a good approach? If not could you think of any alternatives?

Reply:
I wanted to empty the dependency array to call the fetchCats function only once. The alternative is to define the function inside the useEffect and call it there:

```
useEffect(() => {
const fetchCats = async() => {
...
}
fetchCats();
}
```

Another alternative is to put fetch logic into a reusable hook and use this hook when needed. Something like this:
```const useFetch = (url, requestInit = {}) => {
  const [isLoading, setIsLoading] = useState(true);
  const [data, setData] = useState(null);
  const [error, setError] = useState(null);

  useEffect(() => {
    setIsLoading(true);
    fetch(url, requestInit)
      .then(res => res.json())
      .then(data => {
        setData(data);
        setIsLoading(false);
      })
      .catch(err => setError(err));
  }, [url, requestInit]);

  return { data, isLoading, error };
}

// Use it when needed:

const { data, isLoading, error } = useFetch('path')